GitCommands/Git/GitModule.cs

```diff
@@ -950,11 +950,13 @@ public GitRevision GetRevision(ObjectId? objectId = null, bool shortFormat = fal
             }
             else
             {
-                string message = ProcessDiffNotes(10);
+                string message = ProcessDiffNotes(startIndex: 10);
 
                 // commit message is not re-encoded by git when format is given
-                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding);
-                revision.Subject = revision.Body?.Substring(0, revision.Body.IndexOfAny(new[] { '\r', '\n' })) ?? "";
+                // See also RevisionReader for parsing commit body
+                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding).TrimStart().TrimEnd();
+                int endSubjectIndex = revision.Body.IndexOf('\n');
+                revision.Subject = endSubjectIndex < 0 ? revision.Body : revision.Body.Substring(0, endSubjectIndex).TrimEnd();
```

Comment:
```suggestion
                int? endSubjectIndex = revision.Body?.IndexOf('\n');
                revision.Subject = endSubjectIndex is not >= 0 ? revision.Body : revision.Body.Substring(0, endSubjectIndex).TrimEnd();
```

Reply:
It seems like `ReEncodeCommitMessage()` is incorrectly annotated.
A few other changes are required too.